discord-scripts/invite-management.ts

```diff
@@ -0,0 +1,181 @@
+import { Robot } from "hubot"
+import {
+  Client,
+  TextChannel,
+  VoiceChannel,
+  StageChannel,
+  ApplicationCommandOptionType,
+} from "discord.js"
+
+async function createInvite(
+  discordClient: Client,
+  channelId: string,
+  maxAge: number | string = 86400,
+  maxUses: number | string = 0,
+): Promise<string | null> {
+  if (!discordClient) {
+    throw new Error("Discord client is not initialized.")
+  }
+
+  let channelOrigin: TextChannel | VoiceChannel | StageChannel | null = null
+  try {
+    const fetchedChannel = await discordClient.channels.fetch(channelId)
+    if (
+      fetchedChannel instanceof TextChannel ||
+      fetchedChannel instanceof VoiceChannel ||
+      fetchedChannel instanceof StageChannel
+    ) {
+      channelOrigin = fetchedChannel
+    }
+  } catch (error) {
+    throw new Error("Channel not found or access denied.")
+  }
+
+  if (!channelOrigin) {
+    return null
+  }
+  const numericMaxAge = parseInt(maxAge.toString(), 10)
+  const numericMaxUses = parseInt(maxUses.toString(), 10)
+  if (Number.isNaN(numericMaxAge) || Number.isNaN(numericMaxUses)) {
+    throw new Error("maxAge and maxUses must be valid numbers.")
+  }
+  const invite = await channelOrigin.createInvite({
+    maxAge: numericMaxAge,
+    maxUses: numericMaxUses,
+    unique: true,
+  })
+
+  return invite.url
+}
+
+export default async function sendInvite(discordClient: Client, robot: Robot) {
+  const { application } = discordClient
+
+  if (application) {
+    // Check if create-invite command already exists, if not create it
+    const existingInviteCommand = (await application.commands.fetch()).find(
+      (command) => command.name === "create-invite",
+    )
+    if (existingInviteCommand === undefined) {
+      robot.logger.info("No create-invite command found, creating it!")
+      await application.commands.set([
+        {
+          name: "create-invite",
+          description: "Creates a new invite",
+          options: [
+            {
+              name: "max-age",
+              description: "The maximum age of the invite in hours",
+              type: ApplicationCommandOptionType.Number,
+              required: false,
+            },
+            {
+              name: "max-uses",
+              description: "The maximum uses of the invite",
+              type: ApplicationCommandOptionType.Number,
+              required: false,
+            },
+          ],
+        },
+      ])
+      robot.logger.info("create invite command set")
+    }
+    // create an invite based of the command and channel where the command has been run
+    discordClient.on("interactionCreate", async (interaction) => {
+      if (
+        !interaction.isCommand() ||
+        interaction.commandName !== "create-invite"
+      )
+        return
```

Comment:
Really should have curly braces around this—surprised that prettier didn't slap it actually.

Reply:
Flagging this again.